lib_test/dune

```diff
@@ -1,19 +1,34 @@
-(executables
- (names test_ipaddr test_macaddr test_ppx)
+(rule (copy# ../lib/ipaddr_sexp.ml ipaddr_sexp.ml))
+(rule (copy# ../lib/macaddr_sexp.ml macaddr_sexp.ml))
```

Comment:
Why is this copy needed?

Reply:
This is a trick which come from a previous patch on `ocaml-uri`. The point is:
- tests require `ipaddr.sexp`
- we want to split into two packages `ipaddr` and `ipaddr-sexp`
- `ipaddr-sexp` will depends on `ipaddr`
- tests depends on `ipaddr` and `ipaddr-sexp` because they use some stuffs about `sexp`

So we have a circular dependency where `ipaddr` needs `tests`, which needs `ipaddr-sexp` which needs `ipaddr`.

About `ocaml-uri` (which had the same problem), we split tests into `test` which needs only `uri` and `test-sexp` which needs `uri-sexp`. However, in the case of `ipaddr`, `ipaddr-sexp`/`ipaddr.sexp` is used as an Oracle (eg. `of_sexp (to_sexp t)` in some parts) and it's more tricky to split it.

To avoid this circular dependency, we did this ugly patch to provide `sexp`'s things without the `ipaddr-sexp` dependency. I did not find something better at this stage 😢 .

Reply:
Hm why does `ipaddr` need `tests`?
```
tests
|  --> ipaddr.sexp
     | --> ipaddr
| ---> ipaddr
```

Anyway, sounds like it's a missing feature in `dune`? I'm fine with this workaround, just wondered what was the cause :)